sql/analyzer/replace_sort.go

```diff
@@ -167,23 +197,49 @@ func replaceIdxSortHelper(ctx *sql.Context, scope *plan.Scope, node sql.Node, so
 		}
 		allSame = allSame && same
 	}
-
 	if allSame {
 		return node, transform.SameTree, nil
 	}
-
 	// if sort node was replaced with indexed access, drop sort node
 	if node == sortNode {
 		return newChildren[0], transform.NewTree, nil
 	}
-
 	newNode, err := node.WithChildren(newChildren...)
 	if err != nil {
 		return nil, transform.SameTree, err
 	}
 	return newNode, transform.NewTree, nil
 }
 
+// buildReverseIndexedTable will attempt to take the lookup from an IndexedTableAccess, and return a new
+// IndexedTableAccess with the lookup reversed.
+func buildReverseIndexedTable(ctx *sql.Context, node sql.Node) (*plan.IndexedTableAccess, bool, error) {
```

Comment:
This isn't currently used anywhere, but could be if we ever figure out how to properly reverse the index.